sources/org.osbuild.curl

```diff
@@ -83,55 +83,168 @@ SCHEMA = """
-        return_code = 0
-        for _ in range(10):
-            curl_command = [
-                "curl",
-                "--silent",
-                "--speed-limit", "1000",
-                "--connect-timeout", "30",
-                "--fail",
-                "--location",
-                "--output", checksum,
-            ]
-            if secrets:
-                if secrets.get('ssl_ca_cert'):
-                    curl_command.extend(["--cacert", secrets.get('ssl_ca_cert')])
-                if secrets.get('ssl_client_cert'):
-                    curl_command.extend(["--cert", secrets.get('ssl_client_cert')])
-                if secrets.get('ssl_client_key'):
-                    curl_command.extend(["--key", secrets.get('ssl_client_key')])
-            # url must follow options
-            curl_command.append(url_path)
-
-            curl = subprocess.run(curl_command, encoding="utf-8", cwd=tmpdir, check=False)
-            return_code = curl.returncode
-            if return_code == 0:
-                break
-        else:
-            raise RuntimeError(f"curl: error downloading {url}: error code {return_code}")
-
-        if not verify_file(f"{tmpdir}/{checksum}", checksum):
-            raise RuntimeError(f"checksum mismatch: {checksum} {url}")
-
-        # The checksum has been verified, move the file into place. in case we race
-        # another download of the same file, we simply ignore the error as their
-        # contents are guaranteed to be  the same.
+        result = CurlResult(source=url_path)
         try:
-            os.rename(f"{tmpdir}/{checksum}", f"{directory}/{checksum}")
-        except FileExistsError:
+            for _ in range(10):
+                curl_command = [
+                    "curl",
+                    "--silent",
+                    "--speed-limit", "1000",
+                    "--connect-timeout", "30",
+                    "--fail",
+                    "--location",
+                    "--output", checksum,
+                ]
+                if secrets:
+                    if secrets.get('ssl_ca_cert'):
+                        curl_command.extend(["--cacert", secrets.get('ssl_ca_cert')])
+                    if secrets.get('ssl_client_cert'):
+                        curl_command.extend(["--cert", secrets.get('ssl_client_cert')])
+                    if secrets.get('ssl_client_key'):
+                        curl_command.extend(["--key", secrets.get('ssl_client_key')])
+                # url must follow options
+                curl_command.append(url_path)
+                curl = subprocess.run(curl_command, encoding="utf-8", cwd=tmpdir, check=False)
+                result.return_code = curl.returncode
+                if curl.returncode == 0:
+                    break
+            else:
+                result.state = CurlResult.ERROR_DOWNLOADING
+                raise RuntimeError(f"curl: error downloading {url}: error code")
```

Comment:
Wont this get caught by the outermost `except RuntimeError` that only does `pass`?

Reply:
No, but this will need some rework in any case.